### ISSUE TITLE
experience broken in canary

### ISSUE THREAD

**Issue by paulirish** [arthurvr/timeline-url#9]
(not the extension) but the timeline loading.

@arthurvr can you file a crbug with an example URL to figure out why this broke and what adjustments we need to make

iirc the URL just becomes simpler.

**Comment by paulirish** [comment]
`inspector.html` not `devtools.html`

**Comment by paulirish** [comment]
Interesting. Started hacking on URLs and simplified things quite a bit. 

cc @auchenberg

1. `chrome-devtools://devtools/bundled/inspector.html?remoteFrontendUrl=chrome-devtools://devtools/remote/serve_rev/@198714/inspector.html&loadTimelineFromURL=https://dl.dropboxusercontent.com/u/39519/temp/kenneth-io-Timeline.json`
  * what the extension currently provides
1. `chrome-devtools://devtools/remote/serve_rev/@198714/inspector.html?loadTimelineFromURL=https://dl.dropboxusercontent.com/u/39519/temp/kenneth-io-Timeline.json`
  * killed the first part.  still get loading progress data
1. `https://chrome-devtools-frontend.appspot.com/serve_rev/@198714/inspector.html?loadTimelineFromURL=https://dl.dropboxusercontent.com/u/39519/temp/kenneth-io-Timeline.json`
  * switched to our cloud hosted frontend with canonical URL. at this point the loading progress indicator broke.
1. `http://frontend.chrome-dev.tools/serve_rev/@198714/inspector.html?loadTimelineFromURL=https://dl.dropboxusercontent.com/u/39519/temp/kenneth-io-Timeline.json`
  * using the sexy DNS alias I set up.
1. `https://frontend.chrome-dev.tools/serve_rev/@198714/inspector.html?loadTimelineFromURL=https://dl.dropboxusercontent.com/u/39519/temp/kenneth-io-Timeline.json`
  * just HTTPS'd it.

**Comment by paulirish** [comment]
To pull off the sweet URL that doesn't start with `chrome-devtools://` the host must have CORS enabled for the asset. 

The old legacy `/Public` folder from Dropbox does, but the newer urls like `https://www.dropbox.com/s/8izs6cr7op16h35/kenneth-io-Timeline.json?dl=1` do not have CORS.

The google drive URLs we sorted out in https://github.com/arthurvr/timeline-url/issues/8#issuecomment-96801739 also do not do CORS. 

So for non-CORS assets the best we can do is links like:
  * `chrome-devtools://devtools/remote/serve_rev/@198714/inspector.html?loadTimelineFromURL=https://docs.google.com/uc?id=0B7K1Uipy0SbDTjNERnNEZUphd00&export=download`

<hr>

tl;dr:

* JSON available with CORS: `https://frontend.chrome-dev.tools/serve_rev/@198714/inspector.html?loadTimelineFromURL=<jsonurl>`
  * linkable and clickable
* JSON available without CORS: `chrome-devtools://devtools/remote/serve_rev/@198714/inspector.html?loadTimelineFromURL=<jsonurl>`
  * cannot be clicked, doesnt play well with linkifiers

**Comment by paulirish** [comment]
I'll have a PR coming in a moment for this.

**Comment by auchenberg** [comment]
@paulirish Is there commitment to the ``https://frontend.chrome-dev.tools/serve_rev/@198714/inspector.html?`` url? I think it would be useful for protocol adaptors like https://github.com/Microsoft/IEDiagnosticsAdapter, as the link is clickable. 

Ping https://github.com/Microsoft/IEDiagnosticsAdapter/issues/29

**Comment by paulirish** [comment]
@auchenberg fronted.chrome-dev.tools is just a DNS alias for https://chrome-devtools-frontend.appspot.com   Both are committed to.

However, `chrome-devtools://` protocol is required for remote debugging so I dont think IEDiag will get much use from the chrome-dev.tools URL.

I wish those links were clickable but it is a security issue. :/
